proxy/auth/k8s.go

```diff
@@ -0,0 +1,248 @@
+package auth
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/flightctl/flightctl-ui/bridge"
+	"github.com/flightctl/flightctl-ui/config"
+	"github.com/flightctl/flightctl-ui/log"
+)
+
+type TokenAuthProvider struct {
+	apiTlsConfig *tls.Config
+	authURL      string
+}
+
+type TokenLoginParameters struct {
+	Token string `json:"token"`
+}
+
+func NewTokenAuthProvider(apiTlsConfig *tls.Config, authURL string) *TokenAuthProvider {
+	return &TokenAuthProvider{
+		apiTlsConfig: apiTlsConfig,
+		authURL:      authURL,
+	}
+}
+
+// GetToken does not exist for token auth
+func (t *TokenAuthProvider) GetToken(loginParams LoginParameters) (TokenData, *int64, error) {
+	return TokenData{}, nil, fmt.Errorf("token auth does not use OAuth code flow")
+}
+
+// ValidateToken validates a K8s token by calling the backend API
+func (t *TokenAuthProvider) ValidateToken(token string) (TokenData, *int64, error) {
+	client := &http.Client{Transport: &http.Transport{
+		TLSClientConfig: t.apiTlsConfig,
+	}}
+
+	// Call a basic API endpoint to validate the token
+	validateUrl := config.FctlApiUrl + "/api/v1/fleets?limit=1"
```

Comment:
Is there a better way for the UI to validate the token is accepted on the API side?
cc @asafbennatan @fzdarsky